src/puppetlabs/puppetdb/utils.clj

```diff
@@ -644,3 +643,18 @@
 
 (defn schedule-with-fixed-delay [s f initial-delay delay]
   (.scheduleWithFixedDelay s f initial-delay delay TimeUnit/MILLISECONDS))
+
+(defn pprint-json-parse-exception
+  "Returns a parsed JsonParseException message"
+  [error query]
+  (let [lines (str/split-lines (.getMessage error))
+        positions (re-find #"line: (\d+), column: (\d+)" (lines 1))
+        line-index (Integer. (positions 1))
+        column-index (Integer. (positions 2))
+        error-line-index (- line-index 1)
+        error-column-index (- column-index 2)]
```

Comment:
How did you know we needed to subtract two off the column index to get the `^` in the line underneath to line up with the error in the output? Will this always be the case? It might be useful to add a little more to the doc string of this func to explain how it pulls out the error message and adds a `^` pointer to the location that the JsonParseException is referencing.

Reply:
I subtract 2 from the column index because the indexes returned by the error don't start from 0 and I need to replace the last `space` that I print with the `^` mark.